src/pages/Encounters/tabs/EncounterNotesTab.tsx

```diff
@@ -0,0 +1,373 @@
+  useEffect(() => {
+    if (threadsData?.results.length && !selectedThread) {
+      setSelectedThread(threadsData.results[0].id);
+    }
+  }, [threadsData, selectedThread]);
+
+  // Fetch messages with offset pagination
+  const {
+    data: messagesData,
+    isLoading: messagesLoading,
+    hasNextPage,
+    fetchNextPage,
+    isFetchingNextPage,
+  } = useInfiniteQuery<PaginatedResponse<Message>>({
+    queryKey: ["messages", selectedThread],
+    queryFn: async ({ pageParam = 0 }) => {
+      const response = await query(routes.notes.patient.getMessages, {
+        pathParams: {
+          patientId: encounter.patient.id,
+          threadId: selectedThread!,
+        },
+        queryParams: {
+          limit: String(LIMIT),
+          offset: String(pageParam),
+        },
+      })({ signal: new AbortController().signal });
+      return response as PaginatedResponse<Message>;
+    },
+    initialPageParam: 0,
+    getNextPageParam: (lastPage, allPages) => {
+      const currentOffset = allPages.length * LIMIT;
+      return currentOffset < lastPage.count ? currentOffset : null;
+    },
+    enabled: !!selectedThread,
+  });
+
+  // Add effect for infinite scroll
+  useEffect(() => {
+    if (inView && hasNextPage) {
+      fetchNextPage();
+    }
+  }, [inView, hasNextPage, fetchNextPage]);
+
+  // Create thread mutation
+  const createThreadMutation = useMutation({
+    mutationFn: mutate(routes.notes.patient.createThread, {
+      pathParams: { patientId: encounter.patient.id },
+    }),
+    onSuccess: (newThread) => {
+      queryClient.invalidateQueries({ queryKey: ["threads"] });
+      setNewThreadTitle("");
+      setShowNewThreadForm(false);
+      // Select the newly created thread
+      setSelectedThread((newThread as { id: string }).id);
+    },
+  });
+
+  // Create message mutation
+  const createMessageMutation = useMutation({
+    mutationFn: mutate(routes.notes.patient.postMessage, {
+      pathParams: {
+        patientId: encounter.patient.id,
+        threadId: selectedThread!,
+      },
+    }),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["messages", selectedThread] });
+      setNewMessage("");
+    },
+  });
+
+  const handleCreateThread = () => {
+    createThreadMutation.mutate({
+      title: newThreadTitle,
+      encounter: encounter.id,
+    });
+  };
+
+  if (threadsLoading) {
+    return <Loading />;
+  }
+
+  return (
+    <div className="grid grid-cols-12 gap-4 h-[calc(100vh-12rem)]">
+      {/* Threads List - Make it more compact */}
+      <div className="col-span-3 border-r">
+        <div className="space-y-4 p-4">
+          <div
+            role="button"
+            tabIndex={0}
+            className="flex items-center text-sm hover:text-primary transition-colors cursor-pointer"
+            onClick={() => setShowNewThreadForm(true)}
+          >
+            <Plus className="mr-2 h-4 w-4" />
+            {t("Create New Thread")}
```

Comment:
translation keys missing in en.json, needs to be added as mentioned in Merge Checklist

```suggestion
            {t("create_new_thread")}
```